servers/quarkus-common/src/main/java/org/projectnessie/quarkus/providers/storage/JdbcBackendBuilder.java

```diff
@@ -17,43 +17,63 @@
 
 import static org.projectnessie.quarkus.config.VersionStoreConfig.VersionStoreType.JDBC;
 
-import io.agroal.api.AgroalDataSource;
-import io.quarkus.datasource.common.runtime.DatabaseKind;
+import io.quarkus.arc.All;
+import io.quarkus.arc.InstanceHandle;
 import jakarta.enterprise.context.Dependent;
 import jakarta.inject.Inject;
-import org.eclipse.microprofile.config.inject.ConfigProperty;
+import java.util.List;
+import javax.sql.DataSource;
 import org.projectnessie.quarkus.config.QuarkusJdbcConfig;
+import org.projectnessie.quarkus.config.datasource.DataSourceActivator;
 import org.projectnessie.quarkus.providers.versionstore.StoreType;
 import org.projectnessie.versioned.storage.common.persist.Backend;
 import org.projectnessie.versioned.storage.jdbc.JdbcBackendConfig;
 import org.projectnessie.versioned.storage.jdbc.JdbcBackendFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @StoreType(JDBC)
 @Dependent
 public class JdbcBackendBuilder implements BackendBuilder {
 
-  @SuppressWarnings("CdiInjectionPointsInspection")
-  @Inject
-  AgroalDataSource dataSource;
+  private static final Logger LOGGER = LoggerFactory.getLogger(JdbcBackendBuilder.class);
 
   @Inject
-  @ConfigProperty(name = "quarkus.datasource.db-kind")
-  String databaseKind;
+  @All
+  @SuppressWarnings("CdiInjectionPointsInspection")
+  List<InstanceHandle<DataSource>> dataSources;
 
   @Inject QuarkusJdbcConfig config;
 
   @Override
   public Backend buildBackend() {
-    if (!DatabaseKind.isPostgreSQL(databaseKind) && !DatabaseKind.isH2(databaseKind)) {
```

Comment:
Oversight or intenional removal of this check?

Reply:
How would it interfere with additional DSs as long as the db-kind is supported?

Reply:
Think of someone adding support for, say, Oracle. They'd need to add `&& !DatabaseKind.isOracle(databaseKind)` here, otherwise their application wouldn't start.

Reply:
They'd need to adopt the SQL statements as well.

Reply:
Fair point, I will put the check back then.

Reply:
Hmm sorry but do we need this though? `db-kind` is a build-time property, so for the built-in datasources, we _know_ that the kind is supported. This check will never throw. It would only throw if 1) the server is rebuilt and 2) it is built with a missing extension or driver, or with a wrong `db-kind`.

I mean I don't mind adding it back, but I don't see how it's useful.

Reply:
Ugh - yea - it's build time. But maybe leave it in - it would fail quickly in such a case, so devs do know that something needs to be done.